packages/mongodb-memory-server-core/src/util/MongoBinaryDownload.ts

```diff
@@ -19,6 +19,9 @@ import { RequestOptions } from 'https';
 
 const log = debug('MongoMS:MongoBinaryDownload');
 
+const retryableStatusCodes = [503, 500];
+const retryableErrorCodes = ['ECONNRESET', 'ETIMEDOUT', 'ENOTFOUND', 'ECONNREFUSED'];
```

Comment:
```suggestion
const retryableErrorCodes = ['ECONNRESET', 'ETIMEDOUT', 'ECONNREFUSED'];
```

404 (Not Found) should not be retried, right?

Reply:
ENOTFOUND could happen due to temporary DNS failures.

Reply:
I just got this error in Jenkins CI/CD (docker container builds) using version "mongodb-memory-server": "10.2.0-beta.3" -

```
Downloading MongoDB "8.0.9": 0% (0mb / 95.7mb)
Starting the MongoMemoryServer Instance failed, enable debug log for more information. Error:
 DownloadError: Download failed for url "https://fastdl.mongodb.org/linux/mongodb-linux-x86_64-ubuntu2404-8.0.9.tgz", Details:
aborted
    at IncomingMessage.<anonymous> (/home/scm/workspace/DealCentre/dealcentre-service/dealcentre-service.PR-OPEN/src/node_modules/mongodb-memory-server-core/lib/util/MongoBinaryDownload.js:365:28)
    at IncomingMessage.emit (node:events:518:28)
    at emitErrorNT (node:internal/streams/destroy:170:8)
    at emitErrorCloseNT (node:internal/streams/destroy:129:3)
    at process.processTicksAndRejections (node:internal/process/task_queues:90:21) {
  url: 'https://fastdl.mongodb.org/linux/mongodb-linux-x86_64-ubuntu2404-8.0.9.tgz',
  msg: 'aborted'
}
Starting the MongoMemoryReplSet Instance failed, enable debug log for more information. Error:
 DownloadError: Download failed for url "https://fastdl.mongodb.org/linux/mongodb-linux-x86_64-ubuntu2404-8.0.9.tgz", Details:
aborted
    at IncomingMessage.<anonymous> (/home/scm/workspace/DealCentre/dealcentre-service/dealcentre-service.PR-OPEN/src/node_modules/mongodb-memory-server-core/lib/util/MongoBinaryDownload.js:365:28)
    at IncomingMessage.emit (node:events:518:28)
    at emitErrorNT (node:internal/streams/destroy:170:8)
    at emitErrorCloseNT (node:internal/streams/destroy:129:3)
    at process.processTicksAndRejections (node:internal/process/task_queues:90:21) {
  url: 'https://fastdl.mongodb.org/linux/mongodb-linux-x86_64-ubuntu2404-8.0.9.tgz',
  msg: 'aborted'
}
```